### ISSUE TITLE
ERS.ne4_oQU240.F2010.MACHINE_gnu.eam-hommexx is failing COMPARE

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/E3SM#4957]
The machines I've tried this test with GNU have all failed compare. The DEBUG version passes.
I've tested with master of May6th on cori-haswell:

`/global/cscratch1/sd/ndk/e3sm_scratch/cori-haswell/ERS.ne4_oQU240.F2010.cori-haswell_gnu.eam-hommexx.20220517_122134_hut7b6`

Also same fail with cpu-only nodes of PM that use gnu v11. And I see this test fails on ascent.

**Comment by oksanaguba** [comment]
On anvil_gnu compare passed.

**Comment by oksanaguba** [comment]
I believe the state is that it fails with gnu on PM, ascent, and cori HSW. It passes with debug on PM and HSW. The restart PR (Q init) does not seem to help.

**Comment by ndkeen** [comment]
Yes I also tried adding in the changes from `oksanaguba/homme/print-diagn` but did not see different behavior.

**Comment by ndkeen** [comment]
Using nvidia compiler on pm-cpu also shows this test fails compare. So not necessarily specific to GNU. 
`ERS.ne4_oQU240.F2010.pm-cpu_nvidia.eam-hommexx`

**Comment by oksanaguba** [comment]
Is this the only test that is failing on PM or cori HSW with gnu?

Also, to make it little simpler, would you be able to reproduce this fail on anvil or chrysalis, maybe, by changing version of gnu? Debugging on ANL is easiest.

**Comment by ndkeen** [comment]
The test passes on chrysalis `ERS.ne4_oQU240.F2010.chrysalis_gnu.eam-hommexx` which uses gnu v9.3

**Comment by ambrad** [comment]
@ndkeen does cori have GNU version 8 or 9 available? If so, would you run the test with the lower GNU version to see if we can figure out if this is a GNU version issue or an arch-related issue? Thanks.

**Comment by ndkeen** [comment]
Yes I had assumed that the version of GNU was the issue. Cori does have 8.3, but curiously not 9.

**Comment by ndkeen** [comment]
This test still fails on pm-cpu with July 5th master.

**Comment by rljacob** [comment]
Most of this if from the test suites unless indicated.  Its not just gnu

Fails with:
pgi 21.11 (ascent)
gnu 9.1 (ascent)
ibm 16.1.1 (ascent)
gnu 8.3 (crusher)
gnu 11.2 (perlmutter cpu)
pgi 19.10 (compy)
gnu 10.3 (cori-haswell, manual testing)

Passes:
intel 19.0.3 (cori-knl, cori-haswell)
intel 20.0.4 (chrysalis)
gnu 8.2 (anvil)
gnu 8.1  (mappy)
gnu 9.3 (chrysalis, manual testing)

Gnu only (with architecture)
8.1  PASS (Intel)
8.2 PASS (Intel)
8.3 FAIL (AMD)
9.1 FAIL (Power9)
9.3 PASS  (AMD)
10.3 FAIL (Intel)
11.2 FAIL (AMD)

**Comment by ndkeen** [comment]
`ERS.ne4_oQU240.F2010.cori-knl_gnu.eam-hommexx` still fails with Oct repo.

**Comment by rljacob** [comment]
@ambrad or @oksanaguba any thoughts on this?

**Comment by ambrad** [comment]
This is Osana's task. In the past I have suggested switching this to SMS if this won't be debugged right now.

**Comment by oksanaguba** [comment]
Another option is to convert this test to D, if debug passes on all other machines.

**Comment by oksanaguba** [comment]
I must say I do not remember it passing everywhere. I remember PGI was a problem from the beginning, but I do not remember GNU behavior. It is hard to track tests that do not run everyday and do not have baselines. Also, some of the machines did not run stably at the time corresponding PR went in.

Maybe it is a bug, but this works on some machines and in debug. There was a related issue, when I was debugging forcing logic in hommexx, i tried to make eam+homme and eam+hommexx runs bfb and failed (the verification had to be done via climo runs instead). I assume this is because there is some arrangement in init code that causes nonbfb (standalone homme presumably tests all functional, non-init, code). 

Considering that this is not needed for scream, i would leave this as is. Moving it to D on all machines will remove passing opt. build on chrysalis.

**Comment by oksanaguba** [comment]
What about adding a D test, but keeping the opt. build test too?

**Comment by rljacob** [comment]
But our goal is all-green.  Having a test be red forever is not an option.

**Comment by rljacob** [comment]
Actually I just realized this is passing on gcp with gnu 12 so maybe a compiler upgrade on pm-cpu is what we need.  In that case, this could stay red while waiting for that.

**Comment by ndkeen** [comment]
I just tried the test on alvarez, which is just like pm-cpu and has GNU 12.1
The test still fails compare.

**Comment by ndkeen** [comment]
Noting same compare fail with next of July 18th

**Comment by ndkeen** [comment]
I think currently, it looks this test only failing on pm-cpu/nvidia. Fails compare.
Perhaps we can make bless request ?  I would just have no idea what changed to allow passes on other machines.
`ERS_D.ne4_oQU240.F2010.pm-cpu_nvidia.eam-hommexx`

**Comment by ambrad** [comment]
I think the problem is that it fails restart-compare, not just baseline-compare, so blessing the diffs won't help.

**Comment by ndkeen** [comment]
Ah. Thanks Andrew. It's been a while since I've looked at this and forgot the issue.

**Comment by ndkeen** [comment]
With Oct27th checkout, I still see compare fails
```
ERS.ne4_oQU240.F2010.pm-cpu_gnu.eam-hommexx
ERS.ne4_oQU240.F2010.pm-cpu_nvidia.eam-hommexx
```
